src/api/routers/vertex.py

```diff
@@ -0,0 +1,195 @@
+import httpx
+import json
+import logging
+import os
+import requests
+import uuid
+
+from fastapi import Request, Response
+from contextlib import asynccontextmanager
+from api.setting import API_ROUTE_PREFIX, GCP_PROJECT_ID, GCP_REGION, USE_MODEL_MAPPING
+from google.auth import default
+from google.auth.transport.requests import Request as AuthRequest
+
+from api.modelmapper import get_model
+
+known_chat_models = [
+    "publishers/meta-llama/models/llama-3-1-8b-instruct",
+    "publishers/meta-llama/models/llama-3-8b-instruct",
+    "publishers/mistral-ai/models/mistral-7b-instruct-v0.3",
+    "publishers/mistral-ai/models/mistral-nemo-instruct-2407",
+    "publishers/mistral-ai/models/mistral-nemo@2407",
+    "publishers/mistral-ai/models/mistral-7b-instruct@v0.3",
+    "publishers/google/models/gemma-2-27b-it",
+    "publishers/google/models/gemma-2-9b-it",
+    "publishers/google/models/gemma-2b"
+    "publishers/google/models/gemini-2.0-flash-001",
+    "publishers/google/models/gemini-2.0-flash-lite-001",
+    "publishers/google/models/gemini-2.5-pro-preview-05-06",
+    "publishers/google/models/gemini-2.5-flash-preview-05-20",
+    "publishers/meta/models/llama3-8b",
+    "publishers/meta/models/llama-3-1-8b-instruct",
+    "publishers/meta/models/llama2-7b",
+]
+
+
+# GCP credentials and project details
+credentials = None
+project_id = None
+location = None
+
+def get_gcp_project_details():
+    from google.auth import default
+
+    # Try metadata server for region
+    credentials = None
+    project_id = GCP_PROJECT_ID
+    location = GCP_REGION
+
+    try:
+        credentials, project = default()
+        if not project_id:
+            project_id = project
+
+        if not location:
+            zone = requests.get(
+                "http://metadata.google.internal/computeMetadata/v1/instance/zone",
+                headers={"Metadata-Flavor": "Google"},
+                timeout=1
+            ).text
+            location = zone.split("/")[-1].rsplit("-", 1)[0]
+
+    except Exception:
+        logging.warning(f"Error: Failed to get project and location from metadata server. Using local settings.")
+
+    return credentials, project_id, location
+
+credentials, project_id, location = get_gcp_project_details()
+
+# Utility: get service account access token
+def get_access_token():
+    credentials, _ = default(scopes=["https://www.googleapis.com/auth/cloud-platform"])
+    auth_request = AuthRequest()
+    credentials.refresh(auth_request)
+    return credentials.token
+
+def get_gcp_target(model, path):
+    """
+    Check if the environment variable is set to use GCP.
+    """
+    if os.getenv("PROXY_TARGET"):
+        return os.getenv("PROXY_TARGET")
+    elif model in known_chat_models and "chat/completions" in path:
```

Comment:
Let's use a stronger condition than any substring
```suggestion
    elif model in known_chat_models and path.endswith("/chat/completions")
```